sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -3,92 +3,78 @@ package certificates
 }
 
-// NewLocalScannerCertificateRequester creates a new local scanner certificate requester that communicates through
-// the specified channels and initializes a new request ID for reach request.
-// To use it call Start, and then make requests with RequestCertificates, concurrent requests are supported.
-// This assumes that the returned certificate requester is the only consumer of `receiveC`.
-func NewLocalScannerCertificateRequester(sendC chan<- *message.ExpiringMessage,
-	receiveC <-chan *central.IssueLocalScannerCertsResponse) Requester {
+type MsgToCentralFn func(ctx context.Context, msg *message.ExpiringMessage) error
+
+// NewLocalScannerCertificateRequester creates a new certificate requester for Local Scanner certificates
+// (Scanner V2 and Scanner V4).
+func NewLocalScannerCertificateRequester(msgToCentralFn MsgToCentralFn) Requester {
 	return newRequester[
 		*central.IssueLocalScannerCertsRequest,
 		*central.IssueLocalScannerCertsResponse,
 	](
-		sendC,
-		receiveC,
 		&localScannerMessageFactory{},
-		&localScannerResponseFactory{},
 		nil,
+		msgToCentralFn,
 	)
 }
 
-// NewSecuredClusterCertificateRequester creates a new certificate requester that communicates through
-// the specified channels and initializes a new request ID for reach request.
-// To use it call Start, and then make requests with RequestCertificates, concurrent requests are supported.
-// This assumes that the returned certificate requester is the only consumer of `receiveC`.
-func NewSecuredClusterCertificateRequester(sendC chan<- *message.ExpiringMessage,
-	receiveC <-chan *central.IssueSecuredClusterCertsResponse) Requester {
+// NewSecuredClusterCertificateRequester creates a new certificate requester for Secured Cluster certificates.
+func NewSecuredClusterCertificateRequester(msgToCentralFn MsgToCentralFn) Requester {
 	return newRequester[
 		*central.IssueSecuredClusterCertsResponse,
 		*central.IssueSecuredClusterCertsResponse,
 	](
-		sendC,
-		receiveC,
 		&securedClusterMessageFactory{},
-		&securedClusterResponseFactory{},
 		func() *centralsensor.CentralCapability {
 			centralCap := centralsensor.CentralCapability(centralsensor.SecuredClusterCertificatesReissue)
 			return &centralCap
 		}(),
+		msgToCentralFn,
 	)
 }
 
 func newRequester[ReqT any, RespT protobufResponse](
-	sendC chan<- *message.ExpiringMessage,
-	receiveC <-chan RespT,
 	messageFactory messageFactory,
-	responseFactory responseFactory[RespT],
 	requiredCentralCapability *centralsensor.CentralCapability,
+	msgToCentralFn MsgToCentralFn,
 ) *genericRequester[ReqT, RespT] {
 	return &genericRequester[ReqT, RespT]{
-		sendC:                     sendC,
-		receiveC:                  receiveC,
 		messageFactory:            messageFactory,
-		responseFactory:           responseFactory,
+		responseReceived:          concurrency.NewSignal(),
 		requiredCentralCapability: requiredCentralCapability,
+		msgToCentralFn:            msgToCentralFn,
```

Comment:
(solution to the blocker)  Here we would rather do
```suggestion
		messagesToCentral:            make(chan *message.ExpiringMessage),
```